components/bsx/input/TokenBalanceInput.vue

```diff
@@ -7,40 +7,31 @@
     expanded />
 </template>
 
-<script lang="ts">
+<script setup lang="ts">
 import BasicBalanceInput from '@/components/shared/form/BasicBalanceInput.vue'
-import AssetMixin from '@/utils/mixins/assetMixin'
-import { Component, Prop, VModel, mixins } from 'nuxt-property-decorator'
 
-@Component({
-  components: {
-    BasicBalanceInput,
-  },
-})
-export default class TokenBalanceInput extends mixins(AssetMixin) {
-  @VModel() vValue!: string | number
-  @Prop({ type: String, required: false, default: '5' }) tokenId!:
-    | string
-    | number
-  // @Prop({ type: Number, default: '0' }) min!: number
-  // @Prop({ type: Number, default: undefined }) max!: number
+const balanceInputComponent = ref<BasicBalanceInput>()
 
-  get asset() {
-    return this.assetIdOf(this.tokenId)
-  }
+const emit = defineEmits(['input'])
 
-  get unit() {
-    return this.asset.symbol
-  }
+const props = defineProps({
+  value: { type: String || Number, required: true },
+  tokenId: { type: String || Number, required: false, default: '5' },
+})
 
-  get decimals() {
-    return this.asset.decimals
-  }
+const vValue = computed({
+  get() {
+    return props.value
+  },
+  set(value) {
+    emit('input', value)
+  },
+})
```

Comment:
```suggestion
const vValue = computed({
  get: () => props.value,
  set: (value) => {
    emit('input', value)
  },
})
```

Reply:
hi @roiLeo, I would love to use this but our style guide prevents me from doing so.
<img width="834" alt="image" src="https://github.com/kodadot/nft-gallery/assets/3810177/1793d570-ce8d-4a22-a134-fa553ab67c36">

I would really prefer the shorter syntax, would you want me to update the style guide to make this a standard?

Reply:
> hi @roiLeo, I would love to use this but our style guide prevents me from doing so.

This is not related, we use shortand computed on vue files.

Reply:
Noted on this 😊